### ISSUE TITLE
Table cells and their buttons are not accessible via keyboard or screen reader

### ISSUE THREAD

**Issue by rogermparent** [iterative/vscode-dvc#1493]
Originally diagnosed in https://github.com/iterative/vscode-dvc/pull/1461#discussion_r833674775

Currently, the `FirstCell` and `RowContent` components have `aria-hidden=true` on them, likely to get around the [click-events-have-key-events](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/18266282424c546f47a74f842a7bf62bdde1d5e8/docs/rules/click-events-have-key-events.md) ESLint rule.

While this does make the rule pass, it has some obvious downsides:
- These components and their children are not accessible by screen readers, and by extension the default settings of `testing-library`'s `ByRole` queries.
- It defeats the whole reason the ESLint rule is in place, as these clickable elements are not accessible via keyboard.

While the tests in #1461 were fixed by making the element always render in a Portal outside of the table itself, we should probably address this both for accessibility best practices (even if most other parts and to minimize the possibility of `ByRole` confusion in the future.

To implement this fix, we could either:
- Add click handlers to the divs that currently have `aria-hidden: true` that act like buttons
- Change the current click handlers into `button` elements to take advantage of native click handling

I'm always preferential toward the latter option, but turning the whole `RowContent` component into a `button` is a but clunky and heavy-handed. That said, #1461 adding a click action to the cells means we'll probably want to move the click handler off of `RowContent` and onto the bullet indicator which could much more easily be turned into a button.
